smartcontractkit/external-adapters-js
# Auto-generate Adapter Master List

## Closes [#31401](https://app.shortcut.com/chainlinklabs/story/31401/generate-master-ea-list-as-github-workflow)

## Description
This script accomplishes auto-updates of the EA Master List within the Changeset PRs, as well as `composites/README.md`, `sources/README.md`, and `targets/README.md`. The script can be manually run with `yarn generate:master-list` to see changes ahead of time, and you can add `-v` to see verbose logging for errors (which result in `unknown` in the master list).

## Changes
- Add `generate-master-list` script, which will generate 4 markdown documents that describe the current state of all adapters
- Move code that is shared by this and the README generator into a `shared` folder
- Run `yarn generate:master-list` as a check for all PRs, without changing the contents of the PR itself
- Run `yarn generate:master-list` and commit the changes for Changeset PRs



## Steps to Test
`yarn && yarn setup && yarn generate:master-list -v` (expect to see a lot of error logs, which are usually due to incorrect file structure or issues importing EA endpoints or configs).

## Quality Assurance
